wgpu-core/src/command/transfer.rs

```diff
@@ -157,21 +161,40 @@ pub(crate) fn validate_linear_texture_data(
     buffer_side: CopySide,
     bytes_per_block: BufferAddress,
     copy_size: &Extent3d,
+    need_copy_aligned_rows: bool,
 ) -> Result<BufferAddress, TransferError> {
     // Convert all inputs to BufferAddress (u64) to prevent overflow issues
     let copy_width = copy_size.width as BufferAddress;
     let copy_height = copy_size.height as BufferAddress;
     let copy_depth = copy_size.depth_or_array_layers as BufferAddress;
 
     let offset = layout.offset;
-    let rows_per_image = layout.rows_per_image as BufferAddress;
-    let bytes_per_row = layout.bytes_per_row as BufferAddress;
 
     let (block_width, block_height) = format.describe().block_dimensions;
     let block_width = block_width as BufferAddress;
     let block_height = block_height as BufferAddress;
     let block_size = bytes_per_block;
 
+    let width_in_blocks = copy_width / block_width;
+    let height_in_blocks = copy_height / block_height;
+
+    let bytes_per_row = if let Some(bytes_per_row) = layout.bytes_per_row {
+        bytes_per_row.get() as BufferAddress
+    } else {
+        if copy_depth > 1 || block_height > 1 {
```

Comment:
this should be
```
if copy_depth > 1 || height_in_blocks > 1 {
```

Reply:
very nice catch! That would have broken all compressed single row texture updates 😄